### PR TITLE
vhost-vdpa: fix DeviceInfo file content

### PR DESCRIPTION
**Pull request by lmilleri**
This PR fixes an issue in the DeviceInfo file generation for vhost-vdpa devices.

For details please check the following issue:
https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/issues/492

### CODE REVIEW

**Review by adrianchiris — APPROVED** [review]

**Comment by lmilleri** [comment]
@SchSeba @bn222 Can you please have a look and merge if looks good to you?

**Review by SchSeba — APPROVED** [review]

**Comment by lmilleri** [comment]
@bn222 AFAIK there is no error condition to be handled at the call site. Let's say, if a path is returned, it'll be used to fill some structs, otherwise an empty path will be stored instead. The glog.Infof is just for convenience.

**Comment by bn222** [comment]
It seems to me that whenever this function returns "", it was due to it being called on something that doesn't have this path, i.e. it should not have been called on it the first place?

**Comment by lmilleri** [comment]
Added error handling for GetPath function

**Review by e0ne — APPROVED** [review]
LGTM

File: pkg/infoprovider/vdpaInfoProvider.go [hunk 1]
```diff
@@ -60,7 +60,11 @@ func (vip *vdpaInfoProvider) GetDeviceSpecs() []*pluginapi.DeviceSpec {
 
 	// DeviceSpecs only required for vhost vdpa type as the
 	if vip.vdpaType == types.VdpaVhostType {
-		vdpaPath := vip.dev.GetPath()
+		vdpaPath, err := vip.dev.GetPath()
+		if err != nil {
+			glog.Errorf("Unexpected error when fetching the vdpa device path: %s", err)
+			return devSpecs
```
Comment (adrianchiris) [inline pkg/infoprovider/vdpaInfoProvider.go:66]: shouldn't we return nil like above ? (When device not healthy)

Comment (lmilleri) [reply pkg/infoprovider/vdpaInfoProvider.go:66]: I think it would be safer, will do the change thanks!

**Pull request merged**